### PR TITLE
Adding storage feature support

### PR DESCRIPTION
**Pull request by mclacore**
Adding support for premium storage accounts (that do not allow queue_properties to deploy). Premium storage is excellent for low latency/high bandwidth/high transaction rate types of storage. Ideal scenarios include AI/ML storage.

Tested locally w/ azure data lake storage bundle

### CODE REVIEW

File: azure/storage-account/main.tf [hunk 1]
```diff
@@ -18,13 +18,28 @@ resource "azurerm_storage_account" "main" {
...
+        read                  = var.queue_properties.logging.read
+        write                 = var.queue_properties.logging.write
+        version               = var.queue_properties.logging.version
+        retention_policy_days = var.queue_properties.logging.retention_policy_days
+      }
+    }
+  }
+
+  dynamic "blob_properties" {
+    for_each = var.enable_blob ? [1] : []
```
Comment (chrisghill) [inline azure/storage-account/main.tf:35]: Same as above.

**Review by chrisghill — CHANGES_REQUESTED** [review]
A few comments.

File: azure/storage-account/main.tf (hunk 1 above)
Comment (coryodaniel) [reply azure/storage-account/main.tf:35]: I have a similar code block in aurora and had to use a string for I assume the same dynamic-ish single input concept...

https://github.com/massdriver-cloud/aws-aurora-postgresql/blob/56806b2df11dc939a51fcc19eec5dc2ce0fd9b4a/src/cluster.tf#L39

File: azure/storage-account/variables.tf [hunk 2]
```diff
@@ -51,6 +51,35 @@ variable "enable_data_lake" {
   default = false
 }
 
+variable "enable_queue" {
```
[the remark this answers is not in the document]
Comment (mclacore) [reply azure/storage-account/variables.tf:54]: if `enable_queue` is true, but `queue_properties` is empty, then it errors out saying that the values are required but missing. If `enable_queue` is false, and `queue_properties` is populated, then it deploys, but the `queue_properties` is null in the statefile.

**Review by chrisghill — APPROVED** [review]

**Pull request merged**